include/sway/input/keyboard.h

```diff
@@ -21,7 +21,9 @@ struct sway_shortcut_state {
 	 * including duplicates when a keycode generates multiple key ids.
 	 */
 	uint32_t pressed_keycodes[SWAY_KEYBOARD_PRESSED_KEYS_CAP];
-	int last_key_index;
+	uint32_t last_keycode;
+	uint32_t last_raw_modifiers;
+	int npressed;
```

Comment:
It's not worth the type disagreement with the `int` length parameter from `list_t`, and I would rather avoid unsigned wherever a difference might be computed so as to avoid future type conversion or state bugs that trigger effectively infinite loops.

(CONTRIBUTING.md points to [Linux kernel coding style](https://www.kernel.org/doc/Documentation/process/coding-style.rst) which says nothing, [Google Style](https://google.github.io/styleguide/cppguide.html#Integer_Types) votes for signed, most sway code has chosen `int` in similar cases, and wlroots uses `uint32_t` to signal nonnegativity.)

Reply:
`size_t` is the standard type for representing sizes and array indices. I would consider the fact that `list_t` uses `int` to be a bug.
wlroots uses `uint32_t` to represent protocol values, which really are 32 bit unsigned values.

Reply:
To be pedantic, `npressed` is a count, not a size ;-). Still, it would be nice if CONTRIBUTING.md were to mention this convention.

Reply:
>It's not worth the type disagreement with the int length parameter from list_t

Yeah, `list_t.length` being an `int` is a mistake. However you don't use `list_t` so this shouldn't be an issue.

>To be pedantic, npressed is a count, not a size

This important thing is that `size_t` is guaranteed to hold an array index.

Note that as @ascent12 said not all unsigned values should be `size_t`.